pkg/freeze/crio/pause.go

```diff
@@ -0,0 +1,85 @@
+package crio
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+
+	"google.golang.org/grpc"
+
+	"knative.dev/container-freezer/pkg/freeze/common"
+)
+
+const defaultCrioAddress = "/var/run/crio/crio.sock"
+
+func NewCrioProvider() (*CrioCRI, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, defaultCrioAddress, grpc.WithInsecure(), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(1024*1024*16)), grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+		return (&net.Dialer{}).DialContext(ctx, "unix", addr)
+	}))
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", defaultCrioAddress)
+			},
+		},
+	}
+
+	return &CrioCRI{conn: conn, crio: client}, nil
+}
+
+type CrioCRI struct {
+	conn *grpc.ClientConn
+	crio *http.Client
+}
+
+// List returns a list of all non queue-proxy container IDs in a given pod
+func (c *CrioCRI) List(ctx context.Context, podUID string) ([]string, error) {
+	containerIDs, err := common.List(ctx, c.conn, podUID)
+	return containerIDs, err
+}
+
+// Pause performs a pause action on a specific container
+func (c *CrioCRI) Pause(ctx context.Context, container string) error {
+	resp, err := c.crio.Get("http://localhost/pause/" + container)
```

Comment:
Just want to double check that `localhost` is correct here

Reply:
I am doing e2e test, will know the result

Reply:
yeah, it's this path. I tested it.